### ISSUE TITLE
Policy templates vs Group Policies

### ISSUE THREAD

**Issue by Cade66** [cadeyrn/enterprise-policy-generator#135]
Hi,

Enterprise Policy Generator 5.1.0 / Firefox 76.0.1
Mozilla Firefox Group Policies in `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Mozilla`

NOTE 1 : I added only "DisableDefaultBrowserAgent" to registry-policies.reg because
this template doesn't appear in 'Enterprise Policy Generator' 5.1.0 Firefox extension.

NOTE 2 : I added only "DisableMasterPasswordCreation" to registry-policies.reg because
this template IS NOT FUNCTIONAL  in 'Enterprise Policy Generator' 5.1.0 Firefox extension.

My registry-policies.reg file:

```
Windows Registry Editor Version 5.00
; Firefox Policies
[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Mozilla\Firefox]
"DisableDefaultBrowserAgent"=dword:00000001
"DisableMasterPasswordCreation"=dword:00000001
```

It's less Enterprise Policy Generator than Policy templates which is concerned.
I can understand that "DisableDefaultBrowserAgent" is not included yet given its youth, but I linger to know why "DisableMasterPasswordCreation" fails as a Policy template but not as a Group Policy item.

"DisableMasterPasswordCreation" when performed by the Policy template, fails given about:preferences#privacy / Logins and passwords is totally dimmed and Master password inaccessible when performed by the Group Policy but not (& Master Password feasible) when processed by Policy template.

What bothers me also from there on is that I get to wonder if other policy templates fail without me having tested viability of them all.

**Comment by cadeyrn** [comment]
Hi,

I can't really say anything about GPO because I don't use Windows at all. But regarding the two mentioned policies:

DisableDefaultBrowserAgent will be supported in Enterprise Policy Generator 6.0, to be released this month. If DisableMasterPasswordCreation doesn't work it's a bug in Firefox itself because the Enterprise Policy Generator only generates the .json file and the JSON output seems to be correct according to the documention. But maybe it was fixed in the meantime. Is this still an issue for you?

**Comment by Cade66** [comment]
Hi,

I cannot say if the issue I had encountered and detailed above concerning _DisableMasterPasswordCreation_ when set with _policies.json_ is or not fixed with latest Firefox 79.0 because I don't use _policies.json_ any more (be it by setting it manually or with the _Enterprise Policy Generator_ extension). I know **no** issue ever encountered with either _Autoconfig_ or _GPO_.

In my situation using the _Enterprise Policy Engine_ is conducted within the following scheme:

**Autoconfig** : config-prefs.js & config.js (`https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig`),
**Policy templates** (`https://github.com/mozilla/policy-templates/blob/master/README.md`)

Running Firefox, I refer only to policies appearing in its `about:policies#documentation`.
Clicking on a given _Policy Name_ leads to is description on `https://github.com/mozilla/policy-templates/blob/master/README.md#[POLICYNAME]`
There I'll find for the chosen Policy name:

- **Preferences Affected** : if available I'll add them to _Autoconfig_'s config.js otherwise to _GPO_ (i.e. _DisableDefaultBrowserAgent_)
- **Windows (GPO)** : basically used only for templates having no Preferences equivalence.
- **policies.json** (manual or _Enterprise Policy Generator_ extension)

I understand the pertinence of the _Enterprise Policy Generator_ extension which makes it easy to use the _Enterprise Policy Engine_ but in my case, using Windows,  I switched to GPO rather than policies.json first because of the above-mentioned issue then because using a .reg file to set, modify, remove the policies appears to be easy as well (except that user must be aware that removing a setting in the .reg file will require deleting the whole of the GPO before reinstalling it with the new .reg file).

Thanks.

**Comment by cadeyrn** [comment]
Thanks for the answer. Since you're no longer using the Enterprise Policy Generator I will close the issue. 😊
